// ngx-loading/esm2020/lib/ngx-loading.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxLoadingComponent } from './ngx-loading.component';
import * as i0 from "@angular/core";
export class NgxLoadingModule {
    static forRoot(loadingConfig) {
        return {
            ngModule: NgxLoadingModule,
            providers: [{ provide: 'loadingConfig', useValue: loadingConfig }]
        };
    }
}
NgxLoadingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxLoadingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxLoadingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxLoadingModule, declarations: [NgxLoadingComponent], imports: [CommonModule], exports: [NgxLoadingComponent] });
NgxLoadingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxLoadingModule, imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxLoadingModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxLoadingComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        NgxLoadingComponent
                    ]
                }]
        }] });
